<launch>
  <!-- list of arguments that can be given as inputs of the launch file -->
  <arg name="quiet"     default="false" />
  <arg name="filter"    default="geo" />
  <arg name="subcloud"  default="geo" />
  <arg name="plane"     default="ran" />
  <arg name="file"      default="default" />
  <arg name="enable_normal_smoothing" default="true" />
  <arg name="normal_smoothing_alpha" default="0.1" />   <!-- [0,...,1] higher: more responsive, lower: smoother -->
  <arg name="smoothing_cutoff_freq" default="0.2" />    <!-- cut off freq for gaussian kernel smoothing: 1Hz means signals with trends that take >1s to complete cycle pass while filtering out higher freq. changes  -->
  <arg name="use_gaussian_smoothing" default="true" />
  <arg name="enable_scoring" default="true"/>     <!-- bool to disable scoring of normal vector and fallback mechanism-->

  <!-- C++ Executables /-->
  <node name="ground_finder_node" pkg="ground_finder" type="ground_finder_node" output="screen">
    <!-- Parameters -->
    <param name="quiet" value="$(arg quiet)" type="bool"/>
    <param name="filter" value="$(arg filter)" type="string"/>
    <param name="subcloud" value="$(arg subcloud)" type="string"/>
    <param name="plane" value="$(arg plane)" type="string"/>
    <param name="file" value="$(arg file)" type="string"/>
    <param name="enable_normal_smoothing" type="bool" value="$(arg enable_normal_smoothing)" />
    <param name="normal_smoothing_alpha" type="double" value="$(arg normal_smoothing_alpha)" />
    <param name="smoothing_cutoff_freq" type="double" value="$(arg smoothing_cutoff_freq)" />
    <param name="use_gaussian_smoothing" type="bool" value="$(arg use_gaussian_smoothing)" />
    <param name="weight_visibility" type="double" value="0.6" />
    <param name="weight_inlier_ratio" type="double" value="0.4" />
    <param name="score_threshold" value="0.2" type="double" />
    <param name="min_score_sliding_window" value="0.3" type="double "/>
    <param name="enable_scoring" value="$(arg enable_scoring)" type="bool"/>
  </node>

</launch>
